WalletWasabi/WabiSabi/Crypto/ZeroCredentialPool.cs

```diff
@@ -0,0 +1,58 @@
+using System.Linq;
+using System.Collections.Generic;
+using WalletWasabi.Crypto.ZeroKnowledge;
+using System.Collections.Concurrent;
+
+namespace WalletWasabi.WabiSabi.Crypto
+{
+	/// <summary>
+	/// Keeps a registry of credentials available.
+	/// </summary>
+	public class ZeroCredentialPool
+	{
+		private readonly ConcurrentQueue<Credential> _zeroValueCredentials = new ();
+
+		internal ZeroCredentialPool()
+		{
+		}
+
+		/// <summary>
+		/// Registers those zero value credentials that were issued and return the valuable credentials.
+		/// </summary>
+		/// <param name="credentials">Credentials received from the coordinator.</param>
+		public IEnumerable<Credential> RegisterZeroValueCredentials(IEnumerable<Credential> credentials)
+		{
+			foreach (var credential in credentials)
+			{
+				if (credential.Amount.IsZero)
+				{
+					_zeroValueCredentials.Enqueue(credential);
+				}
+				else
+				{
+					yield return credential;
+				}
+			}
+		}
+
+		public Credential[] CompleteWithZeroCredentials(IEnumerable<Credential> credentials)
```

Comment:
Concept ACK but again, I was having a hard time understanding what is this doing because of the naming. I thought complete is about to finish or wait for the end of something. 

- FillOutWithZeroCredentials
- AddZeroCredentialsIfNeeded

Reply:
Renamed as `FillOutWithZeroCredentials`